docs/docs/development/hardware-integration/physical-layouts.md

```diff
@@ -0,0 +1,351 @@
+        transform = <&default_transform>;
+        kscan = <&kscan0>;
+        keys = // List of key positions, optional
+    };
+
+    alt_layout: alt_layout {
+        compatible = "zmk,physical-layout";
+        display-name = "Alternate Layout";
+        transform = <&alt_transform>;
+        kscan = <&alt_kscan0>;
+        keys = // List of key positions, optional
+    };
+};
+```
+
+### Position Map
+
+A position map is used for precise mapping between layouts. It is used to allow [ZMK Studio](../../features/studio.md) to accurately retain keymaps when switching layouts. A keyboard's position map has a child node for every potential layout of the keyboard, with the corresponding entries in `positions` property will be used to determine the mapping between layouts. A position map looks something like this:
+
+Position maps are optional; if a position map is not present, [ZMK Studio](../../features/studio.md) will automatically determine a (potentially inaccurate) mapping based on the physical key properties. This approach is also used as a fallback for positions not specified in the map if the `complete` property is not set.
+
+See also the [configuration section on position maps](../../config/layout.md#physical-layout-position-map).
+
+:::info
+
+Key positions in the maps are numbered like the keys in your keymap, starting at 0. So the first position in the layout is position `0`, the next key position is `1`, etc.
+
+:::
+
+#### Writing a position map
+
+Start by creating the parent node defining the position map:
+
+```dts
+/ {
+    keypad_lossless_position_map {
+        compatible = "zmk,physical-layout-position-map";
+        complete; // Not necessary, but generally recommended if you use position maps
+
+        // Child node 1 here
+
+        // Child node 2 here
+
+        // ...
+    };
+};
+```
+
+Next, select a layout to be the "reference" layout. It is recommended that this be the layout which has the highest number of keys, as that prevents key mappings from being "lost" when switching back and forth between layouts.
+
+Create the child node for the "reference" layout, and fill the `positions` array by iterating through the keys in your layout in the same order as the matrix transform. For a 2x2 macropad the child node would be
+
+```dts
+macropad {
+    physical-layout = <&macropad_layout>;
+    positions // This is equivalent to `positions = <0 1 2 3>;`, reshaped for readability
+        = < 0  1 >
+        , < 2  3 >;
+};
+```
+
+Next, you'll need to write child nodes for every other layout, deriving the position map from the reference. Start by deleting any keys which aren't present in your layout, then reassign the numbers so that they match the matrix transform. Finally, (if the reference layout has more keys) add numbers back into the "empty"/"spare"/"nonexistent" positions so that both position maps count up to the same number. The below examples show this process more clearly.
+
+#### Example larger position map
```

Comment:
Would it be worth adding an even more basic example as the first one that is not complete?

Reply:
Personally I'm not a fan of non-`complete;` position maps, I think they're more likely to lead to confusion. I'd rather omit in favor of encouraging people to be explicit.

Reply:
Both are valid. iIMHO, and should be documented. Non-complete maps can save a ton of duplication, and not waste flash space for constrained devices. They *do* have a place, and should be documented.